cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -0,0 +1,166 @@
+        """Return the sample sheet path from Housekeeper if is valid and exists."""
+        try:
+            sample_sheet_path: Path | None = self.hk_api.get_sample_sheet_path(flow_cell_id)
+        except HousekeeperFileMissingError:
+            LOG.warning(f"Sample sheet for flow cell {flow_cell_id} not found in Housekeeper")
+            return
+        return self.get_valid_sample_sheet_path(sample_sheet_path)
+
+    def get_sample_sheet_content(self, flow_cell: FlowCellDirectoryData) -> list[list[str]]:
+        """Return the sample sheet content for a flow cell."""
+        lims_samples: list[FlowCellSample] = list(
+            get_flow_cell_samples(
+                lims=self.lims_api,
+                flow_cell_id=flow_cell.id,
+                flow_cell_sample_type=flow_cell.sample_type,
+            )
+        )
+        if not lims_samples:
+            message: str = f"Could not find any samples in LIMS for {flow_cell.id}"
+            LOG.warning(message)
+            raise SampleSheetError(message)
+        return create_sample_sheet_content(flow_cell=flow_cell, lims_samples=lims_samples)
+
+    def create_sample_sheet_file(self, flow_cell: FlowCellDirectoryData) -> None:
+        """Create a valid sample sheet in the flow cell directory and add it to Housekeeper."""
+        sample_sheet_content: list[list[str]] = self.get_sample_sheet_content(flow_cell)
+        if not self.force:
+            self.validator.validate_sample_sheet_from_content(sample_sheet_content)
+        LOG.info(f"Writing sample sheet to {flow_cell.sample_sheet_path.resolve()}")
+        if self.dry_run:
+            click.echo(
+                WriteStream.write_stream_from_content(
+                    file_format=FileFormat.CSV, content=sample_sheet_content
+                )
+            )
+            return
+        WriteFile.write_file_from_content(
+            content=sample_sheet_content,
+            file_format=FileFormat.CSV,
+            file_path=flow_cell.sample_sheet_path,
+        )
+        add_and_include_sample_sheet_path_to_housekeeper(
+            flow_cell_directory=flow_cell.path, flow_cell_name=flow_cell.id, hk_api=self.hk_api
+        )
+
+    def validate(self, sample_sheet_path: Path) -> None:
+        """Validate a sample sheet."""
+        self.validator.validate_sample_sheet_from_file(sample_sheet_path)
+
+    def get_or_create_sample_sheet(self, flow_cell_name: str, bcl_converter: str) -> None:
+        """
+        Ensure that a valid sample sheet is present in the flow cell directory.
+        If a valid sample sheet for the flow cell is present in Housekeeper, the function hard-links
+        it to the flow cell directory. If not and if the flow cell directory has a valid sample
+        sheet, the function adds and includes it to Housekeeper. If neither is present, the
+        function creates a sample sheet for the flow cell and adds and includes it to Housekeeper.
```

Comment:
This docstring repeats the function body line by line, consider if it is beneficial creates clutter

Reply:
reduced it to two lines